modules/file/file.go

```diff
@@ -1478,6 +1516,72 @@ func (r *run) checkHash(f fileEntry, hashtype checkType) {
 	return
 }
 
+func (r *run) checkBytes(f fileEntry) {
+	var (
+		err error
+	)
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("checkBytes() -> %v", e)
+		}
+	}()
+	reader := f.getReader()
+	defer f.Close()
+
+	// Must use the "label,search" for loop iteration because of
+	// map[string]search for parameters
+	for _, search := range r.Parameters.Searches {
+		if !search.isactive {
+			continue
+		}
+
+		for _, c := range search.checks {
+
+			decodedByte := make([]byte, hex.DecodedLen(len(c.bytes)))
+			_, err := hex.Decode(decodedByte, c.bytes)
+			if err != nil {
+				e := fmt.Sprintf("unable to decode byte query: %v ", err)
+				panic(e)
+			}
+			scansize := len(decodedByte) // Get the actual byte slice length
+			blocksize := 4 * (2 << 10)   // 4096, 4k
+
+			var bigbuf = make([]byte, (blocksize + (2 * scansize)))
+			// Read slightly more than 4k so we have a full buffer to scan through
+			// as we move into the for loop
+			_, err = reader.Read(bigbuf)
+			if err != nil && err != io.EOF {
+				panic(err)
+			}
+
+			for err != io.EOF {
+				if bytes.Contains(bigbuf, decodedByte) {
+					// We do find the intended hex values in our scan
+					// Break the loop and return
+					// Not certain exactly how to do that here
+					c.storeMatch(f.filename)
+					break
+				} else { //Explicitly, we do not find the hex value, and have not encountered EOF
+					tempBuf := make([]byte, blocksize)
+					_, err = reader.Read(tempBuf)
+					if err != nil && err != io.EOF {
+						panic(err)
+					}
+					// Take a small slice from bigbuf for sliding window
+					smallBuf := make([]byte, 2*scansize)
+					// Get the last two scan-lengths so as to permit the sliding window through End-of-Read
+					copy(smallBuf, bigbuf[blocksize:])
+					// Now make bigbuf out of two scan-lengths plus the 4k buffer read
+					var bigbuf = make([]byte, (2 * scansize))
+					copy(bigbuf, smallBuf)
+					bigbuf = append(bigbuf, tempBuf...)
```

Comment:
I'm not fully convinced this sliding window algorithm works as intended. It would be nice to summarize the logic in a comment at the top of the function, and have tests that focus on verifying it is implemented correctly.
`copy` and `make` are also expensive functions, and it would be good to call them as rarely as possible. My suspicion is this algorithm can be fine tuned further to reduce the cost of the check.